src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -240,8 +244,10 @@ private struct Entry {
             }
         }
 
-        public bool ContainsKey(TKey key) {
-            return FindEntry(key) >= 0;
+        public bool ContainsKey(TKey key)
+        {
+            TValue unused;
+            return TryGetValue(key, out unused);
```

Comment:
If TValue is a wide struct it will be making an expensive copy.
The same with  TryGetValue when it returns false.